kong/tools/database_cache.lua

```diff
@@ -120,21 +121,42 @@ function _M.all_apis_by_dict_key()
 end
 
 function _M.get_or_set(key, cb)
+  local lock = resty_lock:new("cache_locks", {
```

Comment:
`resty_lock:new` can return an error ("dictionary not found" when missing the shm). It should be handled too.